### PR TITLE
add --colorseven --colorshandicap +/- ranked/unranked (needs server support)

### PR DESCRIPTION
**Pull request by wonderingabout**
@anoek @roy7 @Dorus @windo 

this PR is on top of #166 

as explained in #127 , sometimes in handicap games we end up 
giving handicap stones to weaker (but higher ranked) opponents, 
while our bot is superhuman (cant lose against humans even with 
handicap)
so in that example case, we would want to always give stones to 
bot opponents even if they are higher ranked, and reject game if 
that condition is not fullfilled

this PR addresses the issue expressed and proposes a generalization 
of challenge color rejects for 2 cases : 
- 1- even games
- 2- handicap games

for both cases, a default of allowing all colors is included, similar to how 
boardsizes, komis, speeds, and timecontrols families already work

to work, this PR needs server support to add `notification.color` 
to the challenge object
then we will be able to debug and test if the value of 
`notification.color` for "automatic" and "random" colors is 
`-1` or `null`, and slightly adjust this proposed code based 
on the result
in regards to that last point, this PR has a common point than with #165, 
where we want to know the automatic handicap stones true value 
(and not `-1` which is not a number of stones), except : 
- that here we should not have a bypass of min/max limits,
- and that we can start an initial support by just using an alias for "automatic" 
same as how notification.komi `null` is given the alias "automatic" in argv, 
so this is a mix of automatic handicap and automatic komi common issues, 
and then ideally we would like to have `notification.color.automatic` and 
`notification.color.random` too, so that for example is automatic or random 
color ends up chosing "black", we accept the game if `--colorshandicap black`

to sum up, we need server support for `notification.color` as a starter, 

big thanks :+1:

### CODE REVIEW

**Comment by wonderingabout** [comment]
so to sum up, and ideally,

i hope we can add server support for (forgive if name is incorrect) : 
- `player.ranking (bot ranking detection)
- `notification.color` (challenge color detection)

and ideally : 
- `notification.handicap.automatic` (handicap stones number (0 to 36, , and not `-1`) when challenge handicap is "automatic")
- `notification.color.automatic` (player color ("black" or "white", and not `-1`) when challenge color is "automatic") 
- `notification.color.random`  (player color ("black" or "white", and not `null`) when challenge color is "random") 
- `komi.automatic` (actual komi value (any positive or negative number, and not `null`) when challenge color is "automatic") 

but we still want to keep the old `-1` and `null` values i think, for example allowing "automatic" komi is more powerful than allowing 5.5 if 9x9, 7.5 if 19x19, 0.5 if handi
but when we allow "automatic" komi (`null`) for example, then if the automatic komi has the same value than chosen komi, game is still rejected : for example in a 19x19 non handicap game, automatic komi will be set to 7.5, but if opponent chooses custom komi `7.5` then we'll get challenge rejected, but if he chooses komi "automatic" which will end up being 7.5 too, then game will be accepted, so less powerful currently

i think we want the best of both worlds, so keep current objects and variables as is, but add on top of that the earlier mentionned new properties in challenger object

@anoek @roy7 @Dorus @windo

**Comment by wonderingabout** [comment]
just did git pull from latest devel :+1:

**Comment by wonderingabout** [comment]
same comment as in https://github.com/online-go/gtp2ogs/pull/165#issuecomment-473623130

besides fixing merge conflicts later, we need server support for `notification.color` to advance on this PR @anoek :+1: 

big thanks

**Comment by wonderingabout** [comment]
just did ```git pull upstream devel``` and merge commited, ready to merge but this needs server support for ```notification.color``` @anoek @Dorus @roy7 @windo

**Comment by wonderingabout** [comment]
we should also make a function for the connection.js part of the code : will do when servers supports it :+1: 

for the part in config.js, a function is already ready, will submit it in the coming days i hope :+1:
